### ISSUE TITLE
Add `verify()`

### ISSUE THREAD

**Issue by searls** [testdouble/testdouble.js#4]
Should be able to verify interactions on a test double, as a test spy.

```
testDouble =  require('testdouble').create();

verify(testDouble(5))  // throws error
```

One way to implement this might be to do the [same trick we do for stubs](https://github.com/testdouble/testdouble.js/blob/master/lib/when/remembers-last-invocation.coffee) (remember the last invocation on the double) and then exclude the most recent invocation when doing the verification check. 

Re-use whatever arg-checker might be introduced by #1 so that argument matchers can be used in either stubbing or verifying

**Comment by searls** [comment]
Along with #11 and with an eye for #12, the call count of each test double will be hard to ascertain because it will be invoked during `when…thenReturn`. 

This seems like an argument for always appending to the test double's stubbings array and then subtracting the stubbing count to arrive at the total call count for verification & public API purposes.

**Comment by searls** [comment]
Implemented in https://github.com/testdouble/testdouble.js/commit/8be1a955735d41fbb9567953dbb1e52d06eb322a & landed in 0.0.4
